pkg/cloud/services/network/secondarycidr.go

```diff
@@ -55,7 +64,9 @@ func (s *Service) associateSecondaryCidr() error {
 		return err
 	}
 
-	record.Eventf(s.scope.InfraCluster(), "SuccessfulAssociateSecondaryCidr", "Associated secondary CIDR with VPC %q", *out.CidrBlockAssociation.AssociationId)
+	if out != nil && out.CidrBlockAssociation != nil {
```

Comment:
Do we need this check here?

When we support IPv6, associated CIDR will show up in another field: `Ipv6CidrBlockAssociation`

I'd suggest either add a note saying this check should be modified when IPv6 is supported or check if there is any scenario where SDK returns nil output with nil error. Because we are only making the AssociateCIDR call if secondary CIDR is provided.

Reply:
Actually `out` is a pointer variable returned by SDK and to avoid `nil pointer exception` I have added it. Sure I will add a note `we should modify this check once IPv6 is supported `

Reply:
What I mean is if the error is nil, output probably won't return as nil ever. You can check the documentation if they mention anything. Most of the time, I cannot find this info but you can try.

And only leaving the `out`'s nil check should be enough, even if there is a scenario where error is nil and output is nil.
If `CidrBlockAssociation ` is nil, iteration on it won't happen anyways.

Reply:
My concern is not the nil check on out, it is CidrBlockAssociation check.

Reply:
Yes Sedef, I checked also I am not getting any scenario where out is nil in case of err ==nil . So it makes sense to remove both the checks `out != nil && out.CidrBlockAssociation != nil` . I will add a note to update the check when   IPv6 is supported